3.3/.sti/bin/run

```diff
@@ -1,25 +1,112 @@
 #!/bin/bash
 
-function is_gunicorn_installed() {
-    pip show gunicorn
+# This script uses these environment variables:
+#
+# APP_MODULE: Python dotted path to your WSGI application.
+#             If not provided, tries to find the Python path to a 'wsgi.py' file
+#             in the source tree.
+#             That file is present in Django projects by default.
+# APP_CONFIG: Optional configuration file for gunicorn.
+# APP_FILE:   Optional path to Python script to run your application.
+#             Defaults to 'app.py' if it exists.
+#
+# DISABLE_MIGRATE: if not empty, inhibits execution of 'manage.py migrate'.
+
+BIND_ADDR="0.0.0.0:8080"
+
+function run() {
+  # For SCL enablement
+  source .bashrc
+
+  set -e
+
+  # Try to run application with the strategies below, in precedence order.
+  # The first successful strategy takes over this script's process via exec.
+  # If no strategy succeed, report an error message and terminate.
+  run_django
+  run_gunicorn "$APP_MODULE"
+  run_python_script "${APP_FILE:-app.py}"
+
+  echo "ERROR: don't know how to run your application."
+  echo "Please set either APP_MODULE or APP_FILE environment variables, or create a file 'app.py' to launch your application."
+  return 1
 }
 
-# For SCL enablement
-source .bashrc
+function run_django() {
+  ! pip show -q django && return
+
+  # Find shallowest manage.py script, either ./manage.py or <project>/manage.py
+  local MANAGE_FILE=$(find . -maxdepth 2 -type f -name 'manage.py' -printf '%d\t%P\n' | sort -nk1 | cut -f2 | head -1)
+
+  django_migrate "$MANAGE_FILE"
+
+  # try to use gunicorn
+  run_gunicorn "$APP_MODULE"
+  # or fallback to Django's development server
+  django_runserver "$MANAGE_FILE"
+}
+
+function django_migrate() {
+  [ -n "$DISABLE_MIGRATE" ] && return
+  echo "---> Migrating database ..."
+  django_manage_cmd "$1" migrate --noinput
+}
+
+function django_runserver() {
+  echo "---> Serving application with 'manage.py runserver' ..."
+  echo "---> WARNING: this is NOT a recommended way to run you application in production!"
```

Comment:
I might be wrong, but the way I understand "not the recommended" implies there is one true recommended way, which is not the case, since Django list [many ways](https://docs.djangoproject.com/en/1.8/howto/deployment/) to deploy your app.

Using gunicorn is "one of the recommended ways", using runserver is "not one of the recommended ways", thus "not a recommended..."